scalafmt-core/shared/src/main/scala/org/scalafmt/Scalafmt.scala

```diff
@@ -67,7 +67,7 @@ object Scalafmt {
   ): Formatted.Result = {
     def getStyleByFile(style: ScalafmtConfig) = {
       val isSbt = FileOps.isAmmonite(filename) || FileOps.isSbt(filename) ||
-        FileOps.isMarkdown(filename)
+        FileOps.isMarkdown(filename) || code.startsWith("#!")
```

Comment:
Is this the line that would allow toplevel statements like in sbt? Though Scala 3 only allows toplevel definitions and not statements, I don't think we need to change anything since it's included in the dialect.

Reply:
yes, that's the line. it doesn't change the dialect to "sbt" but simply enables allowTopLevelStatements.

on a different note: perhaps the formatter should probably always allow top level since its job is not to validate code but to format it 😀

Reply:
Maybe? That shouldn't influence the formatting at all. We can use a more inclusive dialect.

Reply:
But anyway, I don't think we need to enable toplevel statements when shebang is present. I doesn't really change the Scala 3 behaviour. That's just 5 five cents 😅

Reply:
ok, let me split these two up.